tsearch.c.inc

```diff
@@ -52,20 +46,20 @@ static struct naive_tsearch_tree **naive_tsearch_traverse(const void *key, struc
 NAIVE_TSEARCH_API
 void *NAIVE_TSEARCH_TFIND(const void *key, void *const *rootp, naive_tsearch_cmp *compar)
 {
-    return *naive_tsearch_traverse(key, (void*)rootp, compar, 0, NULL);
+    return *naive_tsearch_traverse(key, (struct naive_tsearch_tree**)rootp, compar, 0, NULL);
 }
 
 NAIVE_TSEARCH_API
 void *NAIVE_TSEARCH_TSEARCH(const void *key, void **rootp, naive_tsearch_cmp *compar)
 {
-    return *naive_tsearch_traverse(key, (void*)rootp, compar, 1, NULL);
+    return *naive_tsearch_traverse(key, (struct naive_tsearch_tree**)rootp, compar, 1, NULL);
 }
 
 NAIVE_TSEARCH_API
-void *NAIVE_TSEARCH_TDELETE(const void *NAIVE_TSEARCH_RESTRICT key, void **NAIVE_TSEARCH_RESTRICT rootp, naive_tsearch_cmp *compar)
+void *NAIVE_TSEARCH_TDELETE(const void *__restrict key, void **__restrict rootp, naive_tsearch_cmp *compar)
 {
     struct naive_tsearch_tree *parent = NULL;
-    struct naive_tsearch_tree **q = naive_tsearch_traverse(key, (void*)rootp, compar, 0, &parent);
+    struct naive_tsearch_tree **q = naive_tsearch_traverse(key, (struct naive_tsearch_tree**)rootp, compar, 0, &parent);
```

Comment:
I started getting worried when I saw the changes for C++ compatibility, but it does seem that the resulting code is still C compatible.

However, did you consider just telling CMake to compile this file as C always ? It is not obvious why we would need to compile this file as C++, unless maybe MSVC does not compile it at all (since MSVC does not support C99).

Reply:
I made these changes because when using tsearch as header only, it can be included in a c++ source file. This is needed for the test suite which is c++.

Reply:
When compiling as a library, it is still a .c source.